packages/pnpm-sync-lib/src/pnpmSyncCopy.ts

```diff
@@ -93,17 +99,34 @@ export async function pnpmSyncCopyAsync(options: IPnpmSyncCopyOptions): Promise<
 
   const pnpmSyncJsonFolder = path.dirname(pnpmSyncJsonPath);
 
-  //read the .pnpm-sync.json
-  const pnpmSyncJson = JSON.parse(pnpmSyncJsonContents);
+  // read the .pnpm-sync.json
+  const pnpmSyncJson: IPnpmSyncJson = JSON.parse(pnpmSyncJsonContents);
+
+  // verify if the version is outdated
+  const pnpmSyncJsonVersion: string = getPnpmSyncJsonVersion();
+
+  if (pnpmSyncJsonVersion !== pnpmSyncJson.version) {
```

Comment:
How painful is it for an engineer to have to regenerate this file?  For example, suppose there was a small patch-level bugfix to `pnpm-sync` that did not affect the `.pnpm-sync.json` output -- would it be worthwhile to detect that case and avoid forcing a reinstall?  

If so, then perhaps the `.pnpm-sync.json` version should be decoupled from the NPM package version.  For example it could be a simple integer similar to PNPM's `lockfileVersion`. 

(But then people might forget to increment `lockfileVersion` when making important fixes.  So if it's painless or very infrequently updated, then coupling it to the NPM package version certainly simplifies maintenance.)

Reply:
Maybe, we can handle this problem in two phases.
1. In short term (maybe before version 1.0), the pnpm-sync is under active development, the `.pnpm-sync.json` format could be changed rapidly to handle users' feedback, so the version from `package.json` is more nature and easy to maintain. 
2. In long term, when it is stable enough, we can switch to use a more stable identifier (maybe the PNPM's lockfile format version)